cli/command/stack/deploy.go

```diff
@@ -26,26 +28,37 @@ func NewDeployCommand(c cli.Interface) *cobra.Command {
 		Use:     "deploy",
 		Aliases: []string{"up", "start"},
 		Short:   "Deploy a stack with a docker compose v3 file",
-		PreRunE: cli.ExactArgs(1),
+		PreRunE: cli.RangeArgs(0, 1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.name = args[0]
-			return deploy(c)
+			return deploy(c, args)
 		},
 	}
 	cmd.Flags().StringVarP(&opts.file, "compose-file", "c", "", "Path to a Compose v3 file")
 	return cmd
 }
 
-func deploy(c cli.Interface) error {
-	c.Console().Printf("Deploying stack %s using %s\n", opts.name, opts.file)
+func deploy(c cli.Interface, args []string) error {
+	var name string
+	charArr := []string{".", "_"}
+	if len(args) == 0 {
```

Comment:
Let's put validation  code like this in a pkg utility class. We shouldn't be doing any substitution, just checking that it's a valid domain name component and returning an error if not. Basically a valid name is a valid domain name component that can contain letters and digits, is 1 - 63 characters in length, starts and ends with a letter or digit, and can contain sequences of one or more hyphens and underscores in the string (hostnames can't contain underscores, but domain name components can -- a valid hostname is a valid domain name component, even if a valid domain name component isn't a valid hostname). For go, I think a regex like the following would work:

`^[a-zA-Z0-9]{1}$|^[a-zA-Z0-9][a-zA-Z0-9-_]{1,61}[a-zA-Z0-9]$`

I did (non-exhaustive) testing of the regex [here](https://regex-golang.appspot.com/assets/html/index.html) and it seemed good enough for now. You could dig through Docker's code and see what they're doing to validate. In fact, we might just be better off not validating at all and letting Docker reject the name if it's not valid... (this would definitely be the most expedient way for now).